libexec/rbenv-version-file-read

```diff
@@ -14,6 +14,19 @@ if [ -e "$VERSION_FILE" ]; then
   if [ "$version" = ".." ] || [[ $version == */* ]]; then
     echo "rbenv: invalid version in \`$VERSION_FILE'" >&2
   elif [ -n "$version" ]; then
+    if [[ "$VERSION_FILE" == *\.ruby-version ]]; then
+      variant_file="${VERSION_FILE/\.ruby-version/.ruby-variant}"
```

Comment:
Instead of using a regex replacement here, I think it's more reliable if you extract the directory using `dirname` and then looking for .ruby-variant in the resulting directory name. In theory (however unlikely), the path to .ruby-version could contain multiple instances of that string, like this: `/home/foo/projects/my.ruby-version.test/.ruby-version`